server/src/model.ts

```diff
@@ -1203,10 +1306,36 @@ export async function getAssignmentDataFromCode(
 export async function searchPublicContent(query: string) {
   // TODO: how should we sort these?
 
-  let query_words = query.split(" ");
-  let content = await prisma.content.findMany({
+  const query_words = query.split(" ");
+  const content = await prisma.content.findMany({
     where: {
-      AND: query_words.map((qw) => ({ name: { contains: "%" + qw + "%" } })),
+      AND: query_words.map((qw) => ({
+        OR: [
+          { name: { contains: "%" + qw + "%" } },
```

Comment:
Not for this PR, but I was doing a little reading, and queries like "% word %" cannot be sped up by adding indices to the columns. Instead, what we want to do here is add a [full text search](https://dev.mysql.com/doc/refman/8.4/en/fulltext-search.html). That will also change the search results, presumably to something closer to what a user would expect.

Reply:
Yeah, it looks like Prisma has full text search as a [preview feature](https://www.prisma.io/docs/orm/prisma-client/queries/full-text-search), or we could fall back to raw queries.